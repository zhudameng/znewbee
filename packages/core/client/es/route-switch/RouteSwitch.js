function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import { RouteContext } from './context';
import { useRouteComponent } from './hooks';
export function RouteSwitch(props) {
  var _props$routes = props.routes,
      routes = _props$routes === void 0 ? [] : _props$routes;

  if (!routes.length) {
    return null;
  }

  return /*#__PURE__*/React.createElement(Switch, null, routes.map(function (route, index) {
    if (route.type == 'redirect') {
      return /*#__PURE__*/React.createElement(Redirect, {
        key: index,
        to: route.to,
        push: route.push,
        from: route.from,
        path: route.path,
        exact: route.exact,
        strict: route.strict
      });
    }

    if (!route.path && Array.isArray(route.routes)) {
      route.path = route.routes.map(function (r) {
        return r.path;
      });
    }

    return /*#__PURE__*/React.createElement(Route, {
      key: index,
      path: route.path,
      exact: route.exact,
      strict: route.strict,
      sensitive: route.sensitive,
      render: function render(props) {
        return /*#__PURE__*/React.createElement(RouteContext.Provider, {
          value: route
        }, /*#__PURE__*/React.createElement(ComponentRenderer, _objectSpread(_objectSpread({}, props), {}, {
          route: route
        })));
      }
    });
  }));
}

function ComponentRenderer(props) {
  var _props$route;

  var Component = useRouteComponent(props === null || props === void 0 ? void 0 : (_props$route = props.route) === null || _props$route === void 0 ? void 0 : _props$route.component);
  return /*#__PURE__*/React.createElement(Component, _objectSpread({}, props), /*#__PURE__*/React.createElement(RouteSwitch, {
    routes: props.route.routes
  }));
}