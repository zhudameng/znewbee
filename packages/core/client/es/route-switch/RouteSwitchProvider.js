function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import { Spin } from 'antd';
import React from 'react';
import { useRequest } from '../api-client';
import { RouteSwitchContext } from './context';
export function RouteSwitchProvider(props) {
  var children = props.children,
      components = props.components,
      routes = props.routes;
  return /*#__PURE__*/React.createElement(RouteSwitchContext.Provider, {
    value: {
      routes: routes,
      components: components
    }
  }, children);
}
export function RemoteRouteSwitchProvider(props) {
  var _useRequest = useRequest({
    url: 'uiRoutes:getAccessible'
  }),
      data = _useRequest.data,
      loading = _useRequest.loading;

  if (loading) {
    return /*#__PURE__*/React.createElement(Spin, null);
  }

  return /*#__PURE__*/React.createElement(RouteSwitchProvider, _objectSpread(_objectSpread({}, props), {}, {
    routes: (data === null || data === void 0 ? void 0 : data.data) || []
  }));
}