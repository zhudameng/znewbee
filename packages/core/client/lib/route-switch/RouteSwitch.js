"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RouteSwitch = RouteSwitch;

var _react = _interopRequireDefault(require("react"));

var _reactRouterDom = require("react-router-dom");

var _context = require("./context");

var _hooks = require("./hooks");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function RouteSwitch(props) {
  var _props$routes = props.routes,
      routes = _props$routes === void 0 ? [] : _props$routes;

  if (!routes.length) {
    return null;
  }

  return /*#__PURE__*/_react.default.createElement(_reactRouterDom.Switch, null, routes.map(function (route, index) {
    if (route.type == 'redirect') {
      return /*#__PURE__*/_react.default.createElement(_reactRouterDom.Redirect, {
        key: index,
        to: route.to,
        push: route.push,
        from: route.from,
        path: route.path,
        exact: route.exact,
        strict: route.strict
      });
    }

    if (!route.path && Array.isArray(route.routes)) {
      route.path = route.routes.map(function (r) {
        return r.path;
      });
    }

    return /*#__PURE__*/_react.default.createElement(_reactRouterDom.Route, {
      key: index,
      path: route.path,
      exact: route.exact,
      strict: route.strict,
      sensitive: route.sensitive,
      render: function render(props) {
        return /*#__PURE__*/_react.default.createElement(_context.RouteContext.Provider, {
          value: route
        }, /*#__PURE__*/_react.default.createElement(ComponentRenderer, _objectSpread(_objectSpread({}, props), {}, {
          route: route
        })));
      }
    });
  }));
}

function ComponentRenderer(props) {
  var _props$route;

  var Component = (0, _hooks.useRouteComponent)(props === null || props === void 0 ? void 0 : (_props$route = props.route) === null || _props$route === void 0 ? void 0 : _props$route.component);
  return /*#__PURE__*/_react.default.createElement(Component, _objectSpread({}, props), /*#__PURE__*/_react.default.createElement(RouteSwitch, {
    routes: props.route.routes
  }));
}