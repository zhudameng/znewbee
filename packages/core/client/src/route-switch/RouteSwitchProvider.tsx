import { Spin } from 'antd';
import React from 'react';
import { useRequest } from '../api-client';
import { RouteSwitchContext } from './context';
import { RouteSwitchProviderProps } from './types';

export function RouteSwitchProvider(props: RouteSwitchProviderProps) {
  const { children, components, routes } = props;
  return <RouteSwitchContext.Provider value={{ routes, components }}>{children}</RouteSwitchContext.Provider>;
}

export function RemoteRouteSwitchProvider(props: RouteSwitchProviderProps) {
  const { data, loading } = useRequest({
    url: 'uiRoutes:getAccessible',
  });
  if (loading) {
    return <Spin />;
  }
  return <RouteSwitchProvider {...props} routes={data?.data || []} />;
}
