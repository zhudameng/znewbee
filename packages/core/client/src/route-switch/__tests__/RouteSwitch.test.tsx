import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { RouteSwitch } from '../RouteSwitch';
import { RouteSwitchProvider } from '../RouteSwitchProvider';

const Home = () => <h1>Home</h1>;
const About = () => <h1>About</h1>;

describe('route-switch', () => {
  it('case 1', () => {
    const App = () => {
      return (
        <RouteSwitchProvider components={{ Home, About }}>
          <MemoryRouter initialEntries={['/']}>
            <RouteSwitch
              routes={[
                {
                  type: 'route',
                  path: '/',
                  exact: true,
                  component: 'Home',
                },
              ]}
            />
          </MemoryRouter>
        </RouteSwitchProvider>
      );
    };
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
