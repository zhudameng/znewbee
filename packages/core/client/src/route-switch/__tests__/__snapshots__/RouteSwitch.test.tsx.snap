// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`route-switch case 1 1`] = `
<div>
  <h1>
    Home
  </h1>
</div>
`;
